dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_Basics/AutoScaleBasics.cs

```diff
@@ -0,0 +1,247 @@
+
+var dataPoints = await CloudWatchMethods.GetMetricStatisticsAsync(groupName);
+Console.WriteLine("Details for the metrics collected:");
+dataPoints.ForEach(detail =>
+{
+    Console.WriteLine(detail);
+});
+
+// Disable metrics collection.
+Console.WriteLine("Disabling the collection of metrics for {groupName}.");
+success = await AutoScaleMethods.DisableMetricsCollectionAsync(client, groupName);
+
+if (success)
+{
+    Console.WriteLine($"Successfully stopped metrics collection for {groupName}.");
+}
+else
+{
+    Console.WriteLine($"Could not stop metrics collection for {groupName}.");
+}
+
+// Terminate all instances in the group.
+Console.WriteLine("--- Now terminating all instances in the AWS Auto Scaling group ---");
+
+groups.ForEach(group =>
+{
+    // Only delete groups in the AWS AutoScaling group we created.
+    if (group.AutoScalingGroupName == groupName)
+    {
+        group.Instances.ForEach(async instance =>
+        {
+            await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(client, instance.InstanceId);
+        });
+    }
+});
+
+// Once all instances have been terminated, delete the group.
+Console.WriteLine("--- Deleting the Auto Scaling group ---");
+await AutoScaleMethods.DeleteAutoScalingGroupAsync(client, groupName);
+
+// Delete the launch template.
+var deletedLaunchTemplateName = await EC2Methods.DeleteLaunchTemplateAsync(launchTemplateId);
+
+if (deletedLaunchTemplateName == launchTemplateName)
+{
+    Console.WriteLine("Successfully deleted the launch template.");
+}
+
+Console.WriteLine("The demo is now concluded.");
+
+void DisplayDescription()
+{
+    Console.WriteLine("This code example performs the following operations:");
+    Console.WriteLine(" 1. Create an Amazon EC2 launch template.");
+    Console.WriteLine(" 2. Create an AWS Auto Scaling group.");
+    Console.WriteLine(" 3. Shows the details of the new AWS Auto Scaling group");
+    Console.WriteLine("    to show that only one instance was created.");
+    Console.WriteLine(" 4. Enables  metrics collection.");
+    Console.WriteLine(" 5. Updatings the AWS Auto Scaling group to increase the");
+    Console.WriteLine("    capacity to three.");
+    Console.WriteLine(" 6. Describes AWS Auto Scaling groups again to show the");
+    Console.WriteLine("    current state of the group.");
+    Console.WriteLine(" 7. Changes the desired capacity of the AWS Auto Scaling");
+    Console.WriteLine("    group to use an additional instance.");
+    Console.WriteLine(" 8. Shows that there are now instances in the group.");
+    Console.WriteLine(" 9. Lists the scaling activities that have occurred for the group.");
+    Console.WriteLine("10. Displays the Amazon CloudWatch metrics that have");
+    Console.WriteLine("    been collected.");
+    Console.WriteLine("11. Disables metrics collection.");
+    Console.WriteLine("12. Terminates all instances in the AWS Auto Scaling group.");
+    Console.WriteLine("13. Deletes the Auto Scaling group.}");
```

Comment:
Is there an extra curly bracket at the end of the sentence?

Reply:
Yes, there was. Fixed.